assets/js/components/user-input/UserInputSelectOptions.js

```diff
@@ -41,19 +41,46 @@ export default function UserInputSelectOptions( { slug, options, max } ) {
 	const values = useSelect( ( select ) => select( CORE_USER ).getUserInputSetting( slug ) || [] );
 	const [ other, setOther ] = useState( values.filter( ( value ) => ! options[ value ] )[ 0 ] || '' );
 	const { setUserInputSetting } = useDispatch( CORE_USER );
+	const inputRef = useRef();
+	const [ disabled, setDisabled ] = useState( false );
 
 	// Need to make sure that dependencies list always has the same number of elements.
 	const dependencies = values.concat( Array( max ) ).slice( 0, max );
 
 	const onClick = useCallback( ( event ) => {
 		const { target } = event;
-		const { value, checked } = target;
+		const { value, checked, name, type, id } = target;
 
 		const newValues = new Set( [ value, ...values ] );
 		if ( ! checked ) {
 			newValues.delete( value );
 		}
 
+		if ( name === `${ slug }-other` && checked === true ) {
```

Comment:
What's the reason to use `name` here, but `id` further below for what looks to be the same comparison?